### PR TITLE
[JUJU-4328] Inject controllerConfigGetter in agent/meterstatus facade

### PR DESCRIPTION
**Pull request by anvial**
This PR injects controllerConfigService in agent/meterstatus facade. Also, it refactors agent/uniter facade to mock controllerConfigGetter function in unit-tests.

To inject this controllerConfigService (controllerConfigGetter), the PR extract `ControllerConfig()` function from `UniStateShim` in common.UnitStateAPI. To do this there was a need to inject `controllerConfigGetter` in several facades where common APIs used:
- `agent/provisioner`
- `agent/upgrader`
- `client/client`
- `client/machinemanager`
- `client/modelmanager`
- `client/modelupdater`

## Checklist

- [x] Code style: imports ordered, good names, simple structure, etc
- [x] Comments saying why design decisions were made
- [x] Go unit tests, with comments saying what you're testing

## QA steps

```sh
go test github.com/juju/juju/apiserver/facades/agent/meterstatus/... -gocheck.v
go test github.com/juju/juju/apiserver/facades/agent/uniter/... -gocheck.v 
go test github.com/juju/juju/apiserver/facades/agent/provisioner/... -gocheck.v 
go test github.com/juju/juju/apiserver/facades/agent/upgrader/... -gocheck.v 
go test github.com/juju/juju/apiserver/facades/client/client/... -gocheck.v 
go test github.com/juju/juju/apiserver/facades/client/machinemanager/... -gocheck.v 
go test github.com/juju/juju/apiserver/facades/client/modelmanager/... -gocheck.v 
go test github.com/juju/juju/apiserver/facades/client/modelupgrader/... -gocheck.v

juju bootstrap lxd lxd
juju add-model test
juju deploy ubuntu -n 3
```

### CODE REVIEW

File: apiserver/facades/client/machinemanager/machinemanager.go [hunk 1]
```diff
@@ -317,7 +323,7 @@ func (mm *MachineManagerAPI) addOneMachine(p params.AddMachineParams) (*state.Ma
 
 // ProvisioningScript returns a shell script that, when run,
 // provisions a machine agent on the machine executing the script.
-func (mm *MachineManagerAPI) ProvisioningScript(ctx context.Context, args params.ProvisioningScriptParams) (params.ProvisioningScriptResult, error) {
+func (mm *MachineManagerAPI) ProvisioningScript(ctx context.Context, controllerConfigGetter ControllerConfigGetter, args params.ProvisioningScriptParams) (params.ProvisioningScriptResult, error) {
```
Comment (SimonRichardson) [inline apiserver/facades/client/machinemanager/machinemanager.go:326]: This is wrong, you can't pass this here. This should be on the MachingManagerAPI type.

File: apiserver/facades/client/machinemanager/machinemanager.go [hunk 2]
```diff
@@ -327,7 +333,8 @@ func (mm *MachineManagerAPI) ProvisioningScript(ctx context.Context, args params
 	if err != nil {
 		return result, errors.Trace(err)
 	}
-	icfg, err := InstanceConfig(ctx, st, mm.st, args.MachineId, args.Nonce, args.DataDir)
+
+	icfg, err := InstanceConfig(ctx, controllerConfigGetter, st, mm.st, args.MachineId, args.Nonce, args.DataDir)
```
Comment (SimonRichardson) [inline apiserver/facades/client/machinemanager/machinemanager.go:337]:
```suggestion
	icfg, err := InstanceConfig(ctx, mm.controllerConfigGetter, st, mm.st, args.MachineId, args.Nonce, args.DataDir)
```

File: apiserver/facades/schema.json [hunk 3]
```diff
@@ -22831,18 +22831,6 @@
...
-                    "properties": {
-                        "Params": {
-                            "$ref": "#/definitions/ProvisioningScriptParams"
-                        },
-                        "Result": {
-                            "$ref": "#/definitions/ProvisioningScriptResult"
-                        }
-                    },
-                    "description": "ProvisioningScript returns a shell script that, when run,\nprovisions a machine agent on the machine executing the script."
-                },
```
Comment (SimonRichardson) [inline apiserver/facades/schema.json:22845]: This file should never be affected with the changes we're doing.

File: apiserver/facades/client/machinemanager/machinemanager.go (hunk 1 above)
Comment (anvial) [reply apiserver/facades/client/machinemanager/machinemanager.go:326]: Got it. I used it as part of `MachineManagerAPI`, but then I figured out that we use this function directly in unit tests, and I used contollerConfigGetter as a parameter as one of the exits to pass the mocked controllerConfigService in the unit tests.

I'll think of how to sort it out in another way.

Comment (anvial) [reply apiserver/facades/client/machinemanager/machinemanager.go]: fixed

File: apiserver/facades/schema.json (hunk 3 above)
Comment (anvial) [reply apiserver/facades/schema.json]: got it, thx

File: apiserver/facades/client/machinemanager/machinemanager.go (hunk 2 above)
Comment (anvial) [reply apiserver/facades/client/machinemanager/machinemanager.go]: Done

**Review by SimonRichardson — COMMENTED** [review]
Code looks great.

**Comment by hpidcock** [comment]
> Code looks great.

@SimonRichardson was this meant to be an approval?

**Comment by SimonRichardson** [comment]
I need someone to Q&A it.

**Review by manadart — APPROVED** [review]
QA is fine for this.

**Pull request merged**